superset-frontend/plugins/plugin-chart-echarts/src/Treemap/transformProps.ts

```diff
@@ -141,7 +148,17 @@ export default function transformProps(
   };
   const refs: Refs = {};
   const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
-  const numberFormatter = getNumberFormatter(numberFormat);
+  const numberFormatter =
+    getCustomFormatter(
+      buildCustomFormatters(
+        metric,
+        currencyFormats,
+        columnFormats,
+        numberFormat,
+      ),
+      metric,
+    ) ?? getNumberFormatter(numberFormat);
```

Comment:
At the risk of over-abstraction, would it make sense to have yet one more helper that simplifies this call chain? This code block seems to be repeated a few times so abstracting it would make it slightly more DRY and less heavy on the eyes. Something like
```ts
  const numberFormatter =
    getFormatter(
      metric,
      currencyFormats,
      columnFormats,
      numberFormat,
    );
```

Reply:
💯